tests/unit_tests/test_aggregate_temporal.py

```diff
@@ -0,0 +1,235 @@
+    [
+        (
+            {
+                "intervals": [
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-18T00:00:00.000Z" ],
+                ],
+                "reducer": "({data}) => data.length",
+            },
+            False,
+            None,
+        ),
+        (
+            {
+                "reducer": "({data}) => data.length",
+            },
+            True,
+            "MISSING_PARAMETER",
+        ),
+        (
+            {
+                "intervals": False,
+                "reducer": "({data}) => data.length",
+            },
+            True,
+            "NOT_ARRAY",
+        ),
+        (
+            {
+                "intervals": [
+                    [ "INVALID_INTERVAL" ],
+                ],
+                "reducer": "({data}) => data.length",
+            },
+            True,
+            "Invalid temporal extent.",
+        ),
+        (
+            {
+                "intervals": [
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-18T00:00:00.000Z" ],
+                ],
+                "reducer": "({data}) => data.length",
+                "labels": [ "interval1", "interval2" ],
+            },
+            True,
+            "Number of labels must match number of intervals",
+        ),
+        (
+            {
+                "intervals": [
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-18T00:00:00.000Z" ],
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-20T00:00:00.000Z" ],
+                ],
+                "reducer": "({data}) => data.length",
+            },
+            True,
+            "Distinct dimension labels required",
+        ),
+        (
+            {
+                "intervals": [
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-18T00:00:00.000Z" ],
+                ],
+                "reducer": "({data}) => data.length",
+                "dimension": "NON_EXISTENT_DIMENSION",
+            },
+            True,
+            "Dimension not available.",
+        ),
+        (
+            {
+                "intervals": [
+                    [ "2022-03-16T00:00:00.000Z", "2022-03-18T00:00:00.000Z" ],
+                ],
+                "reducer": "({data}) => data.length",
+                "dimension": "bands_name",
+            },
+            True,
+            "Dimension is not of type temporal.",
+        ),
+    ],
+)
+def test_aggregate_temporal_exceptions(aggregate_temporal_process_code, data, scenes, example_input, raises_exception, error_message):
+    additional_js_code_to_run = (
+        load_datacube_code()
+        + f"const cube = new DataCube({data}, 'bands_name', 'temporal_name', true, [], {scenes});"
+    )
+    process_arguments = f"{{...{json.dumps(example_input)}, 'data': cube, 'scenes': {scenes}, 'reducer': {example_input['reducer']}}}"
+    if raises_exception:
+        with pytest.raises(Exception) as exc:
+            run_process(
```

Comment:
Can `test_aggregate_temporal_exceptions` be changed to use `run_input_validation` method?

It turns out there is a major bug with the tests for exceptions discovered in https://github.com/openEOPlatform/openeo-pg-evalscript-converter/pull/68. When that is merged, this test will fail on Windows devices.

The problem is that the `exc.value` (that we check includes `error_message`) is of form: `Command "node -e <entire-code> failed with exit code 1`.  So for as long your `error_message` string is in the source code, the test will pass...

`run_input_validation` correctly checks against `stderr` attribute of `subprocess.CalledProcessError`, but it also simplifies things.